spec/rspec/matchers/built_in/yield_spec.rb

```diff
@@ -628,6 +695,15 @@ def each_arg(*args, &block)
         |         got: ["food", "barn"]
       EOS
     end
+
+    it 'fails when the successively yielded args match the matchers at return time only' do
+      expect { |b|
+        %w[ barn food ].each do |val|
+          _yield_with_args(val, &b)
+          val.delete!(val)
```

Comment:
Probably more obvious. 👍

My guess is it'd be a bit less efficient, but that doesn't matter. I had originally used `val.clear!` here, but that method isn't defined in 1.8 and I found `val.delete!(val)` as a 1.8-compatible alternative.